qiskit/compiler/transpiler.py

```diff
@@ -921,11 +921,15 @@ def _parse_instruction_durations(backend, inst_durations, dt, circuits):
     take precedence over backend durations, but be superceded by ``inst_duration``s.
     """
     if not inst_durations:
-        backend_durations = InstructionDurations()
-        try:
-            backend_durations = InstructionDurations.from_backend(backend)
-        except AttributeError:
-            pass
+        backend_version = getattr(backend, "version", 0)
+        if backend_version <= 1:
```

Comment:
We have added a lot of these in different places. I wonder if there is a better way to handle the versions better than if-else statements. I imagine it would be very tedious to update these code (and may miss some) when we have a newer version of backend.

Reply:
The way the transpile function is structured right now there isn't really a better way to handle this. I think in the near term I'm going to look at rewriting how we pass backend information to the pass managers here and use a `Target` as the single object we pass to the pass manager constructor. As I agree this code is super hard to deal with (not just for this reason, but also because it makes things like #7789 hard too) and very error prone.